numpyro/distributions/discrete.py

```diff
@@ -592,3 +593,76 @@ def mean(self):
     @lazy_property
     def variance(self):
         return (1 - self.gate) * self.rate * (1 + self.rate * self.gate)
+
+
+class GeometricProbs(Distribution):
+    arg_constraints = {'probs': constraints.unit_interval}
+    support = constraints.nonnegative_integer
+    is_discrete = True
+
+    def __init__(self, probs=None, validate_args=None):
+        self.probs = probs
+        super(GeometricProbs, self).__init__(batch_shape=jnp.shape(self.probs),
+                                             validate_args=validate_args)
+
+    def sample(self, key, sample_shape=()):
+        return geometric(key, self.probs, sample_shape + self.batch_shape)
+
+    @validate_sample
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        value, probs = promote_shapes(value, self.probs)
+        probs = jnp.where((probs == 1) & (value == 0), 0, probs)
+        return value * jnp.log1p(-probs) + jnp.log(probs)
+
+    @property
+    def mean(self):
+        return 1. / self.probs - 1.
+
+    @property
+    def variance(self):
+        return (1. / self.probs - 1.) / self.probs
+
+
+class GeometricLogits(Distribution):
+    arg_constraints = {'logits': constraints.real}
+    support = constraints.nonnegative_integer
+    is_discrete = True
+
+    def __init__(self, logits=None, validate_args=None):
+        self.logits = logits
+        super(GeometricLogits, self).__init__(batch_shape=jnp.shape(self.logits),
+                                              validate_args=validate_args)
+
+    @lazy_property
+    def probs(self):
+        return _to_probs_bernoulli(self.logits)
+
+    def sample(self, key, sample_shape=()):
+        return geometric(key, self.probs, sample_shape + self.batch_shape)
+
+    @validate_sample
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
```

Comment:
In NumPyro, you can remove this check (because it is handled in `@validate_sample` decorator).